core/src/main/java/org/fao/geonet/kernel/datamanager/base/BaseMetadataManager.java

```diff
@@ -945,6 +945,10 @@ public Element updateFixedInfo(String schema, Optional<Integer> metadataId, Stri
                     changeDate = metadata.getDataInfo().getChangeDate().getDateAndTime();
                     createDate = metadata.getDataInfo().getCreateDate().getDateAndTime();
                 }
+                else {
+                    env.addContent(new Element("newRecord").setText("true"));
```

Comment:
I would move that out of the `updateDatestamp` if. We may want to do other stuff when it is a new record (eg. remove thumbnail, ...) ? Add the `newRecord` flag in all cases maybe with something like
```java
env.addContent(new Element("newRecord").setText(metadata == null))
```
?

Reply:
ok I will move out the updateDatestamp.